### ISSUE TITLE
Default location of `.gitignore` and `.eleventyignore` files

### ISSUE THREAD

**Issue by zachleat** [11ty/eleventy#229]
.gitignore is assumed to live at the root of the input directory—maybe it shouldn’t? Might make sense to check to see if that lives in the command root instead (more likely to match the repo root)

Relevant docs: https://www.11ty.io/docs/ignores/

Surfaced in #228

**Comment by kleinfreund** [comment]
> `.gitignore` also lives in the root of the input directory—maybe it shouldn’t?

Let’s first clarify terminology. What is the command root and what is the input root?

Assuming I cloned the repository for my site to `~/kleinfreund.de`, I have a `~/kleinfreund.de/.gitignore`,  a `~/kleinfreund.de/.eleventy.js` and would also expect `~/kleinfreund.de/.eleventyignore` to be located there. I call `~/kleinfreund.de` the project root and it sounds like you call it command root (i.e. the place from where you run the `eleventy` command).

Also, the concept of having a dedicated input directory turned out to have more downsides than benefits for me (e.g includes being input-directory-relative which they shouldn’t). This is why I have set the input directory to `'.'` which results in the input directory *being the same* as the project directory.

**TL;DR**: In my opinion, all configuration-type files that are global should reside in the project root directory, not in any subdirectory.

**Comment by zachleat** [comment]
I agree—the current behavior is wrong.

* `~/kleinfreund.de` would be the command/project root (where I’d run eleventy from)
* If I used `--input=src` then `~/kleinfreund.de/src/` would be my input root (where eleventy would search for templates)

The good news is that it’d be somewhat cheap to look in multiple locations for these two files.

**Comment by zachleat** [comment]
In `0.7.0` eleventy will now search for:

`.gitignore` in the:
* project root (directory you ran eleventy from)
* `input` directory (for backwards compatibility, although I’d argue this doesn’t make sense to do any more 🤡)

`.eleventyignore` in the:
* project root (directory you ran eleventy from)
* `input` directory

Ignore entries found in these files will be normalized using the directory they reside in.

For example, `npx eleventy --input=src` with an `src/.eleventyignore` file with the following content…

```
mytemplate1.md
mytemplate2.md
```

…will normalize entries starting with `src` a la `!./src/mytemplate1.md` or  `!./src/mytemplate2.md` and a root directory `./.eleventyignore` will normalize to `!./mytemplate1.md` and `!./mytemplate2.md`
